### ISSUE TITLE
threebot registration and initialization

### ISSUE THREAD

**Issue by xmonader** [threefoldtech/home#490]
Currently this is done by spawning a container exporting its database and importing it with a user seed.

Here we need to agree on the approach and workflow

**Comment by zaibon** [comment]
Previous approach is defined at https://github.com/threefoldtech/home/issues/415#issue-524859887
But let me summarize what is needed and why we implemented like that the first time.

Requirements:
- 3Bot server must use a derivative of the key pair generated by the 3bot login app/3bot connect to encrypt all the data of its user.
- The master key pair/seed can only be stored on the phone of the user.
- A 3Bot can be recovered with only information its seed.
- No third party can have access to the private key of the user at any point in time. this can NEVER happen.
- Getting his own 3bot should be a user friendly process.

Why we implemented like this the first time?

At the time when we implemented registration there were not possibility to set the derivative key inside the 3bot server before its first start. This is because we (as in 3fold) were deploying the 3bot server our self and thus if we would have put the private key of the user inside before starting it, that would mean we know the private key, this is a not go.

Solution was to first boot the 3bot with a randomly generated key pair. Then allow the user to connect to its 3bot over HTTPS. Then once the connection is secured, the user would enter it's seed so the 3bot could generate a derivative of the seed and re-program itself to use it.
This also had the advantage that we would double check the user have their seed saved in the process.
